src/System.Net.Http/src/System/Net/Http/Managed/HttpConnection.cs

```diff
@@ -256,22 +250,30 @@ private Task WriteFormattedInt32Async(int value, CancellationToken cancellationT
                 // CRLF for end of headers.
                 await WriteTwoBytesAsync((byte)'\r', (byte)'\n', cancellationToken).ConfigureAwait(false);
 
-                // Write body, if any
+                // Flush out anything buffered from the request headers to finish sending them.
+                await FlushAsync(cancellationToken).ConfigureAwait(false);
```

Comment:
There's a related issue here, which is that the write buffering in HttpConnection is somewhat questionable.  We're basically buffering all writes up to the write buffer size.  I'm not sure this is the right behavior, especially for this duplex scenario.

That said, I do think that buffering the outgoing headers and initial content write seems important for performance.

So I guess what I'm suggesting is:
(1) Don't buffer content writes
(2) BUT, if the initial content write can fit in the remaining part of the write buffer, then copy it into the buffer so we can do a single write
(3) (Obviously, if there's no content, we need to flush immediately)
(4) If the user cares about sending the request headers without sending any content, they can explicitly do a flush.  This seems kind of unlikely to me (aren't you going to want to send something in the body, to start things off?), but I could be wrong about that.
(4) Obviously, Expect: 100-continue will cause the headers to be flushed as well.